### PR TITLE
Testing the runtime performance metrics of the browsers

### PR DESCRIPTION
**Pull request by axemclion**
> Work in progress. Created this PR to check if the approach I take is correct. 

The tests here only benchmark the time taken to generate the CSS in each of the frameworks. 
This PR adds benchmarks to check the impact of each of the frameworks from the end user's point of view. It calculates the metrics seen in Chrome summary Pie Chart - loading, scripting, rendering, etc.

Results look something like - 

image

To run the rendering benchmarks

1. [download Chromedriver](https://sites.google.com/a/chromium.org/chromedriver/downloads) and start it
2. Run `npm run compile`
3. Run `ITERATIONS=5000 npm run browserBench`

### CODE REVIEW

File: src/run.js [hunk 1]
```diff
@@ -62,15 +67,83 @@ export const runView = (testName, cases) => (
...
-                    console.error(err);
-                    reject(err);
+            fs.writeFileSync(path, html);
+            console.log(`Wrote ${path}`);
+        });
+        resolve(outputDir);
+    })
+);
+
+export const runBrowser = (testName, cases) => {
```
Comment (PepijnSenders) [inline src/run.js]: This function looks a bit messy with some weird comments, can you clean that up? :)

**Comment by PepijnSenders** [comment]
Very nice @axemclion, very interesting results as well! Could you clean up your code a little bit though from all the comments and TODO's thanks!

**Comment by axemclion** [comment]
@PepijnSenders Here is an update. Can you help me understand what cleanup you would like ?

**Comment by PepijnSenders** [comment]
I see you rebased, this looks good! Thanks for contributing @axemclion, this is a really good one.

**Pull request merged**